blog/2023-01-17-mongodb-monitoring.md

```diff
@@ -0,0 +1,297 @@
+
+### Asserts
+ - **Asserts**: The number of asserts encountered by the MongoDB server per second. The following assert types are monitored: 
+	- regular
+	- warning
+	- msg
+	- user
+	- tripwire
+	- rollovers
+
+Asserts are MongoDB's way of detecting errors within the system. They are essentially checks that the system performs to ensure that data is being handled properly. Asserts can be caused by things such as incorrect data formats, incorrect user input, or a bug in the code. By monitoring the number of asserts encountered, it is possible to identify potential problems before they cause a disruption to service.
+
+For example, if the number of regular asserts increases significantly over a short period of time, it could indicate a bug in the code, or a data format issue. If this is the case, further investigation should be done in order to identify the root cause of the problem and prevent it from causing any service disruptions.
+
+### Transactions
+ - **Current Transactions** The number of current transactions on the MongoDB server. The following transaction types are monitored: 
+	- Active
+	- Inactive
+	- Open
+	- Prepared
+
+Poorly designed or overloaded transactions can cause the server to become unresponsive, resulting in application errors or unavailability. Monitoring the number of current transactions can help to identify potential bottlenecks and allow for adjustments to be made in order to improve performance. It can also help to identify rogue transactions that may be consuming excessive resources. By monitoring current transactions, administrators can make informed decisions about how to optimize their environment and prevent potential issues. 
+
+### Clients
+ - **Active Clients**: The number of active clients (readers and writers) connected to the MongoDB server. 
+ 
+Monitoring this metric can help to identify when it is time to scale up the server to meet the demands of the workload. It can also help to identify any potential issues that may arise due to an overload of clients. For example, if the number of active clients is too high, it can result in requests timing out and errors being returned. By monitoring this metric, these issues can be identified and addressed quickly.
+  
+![image](https://user-images.githubusercontent.com/24860547/212876432-d8038c56-1381-478b-9cf1-bb7c8ef78950.png)
+
+ - **Queued Clients**: The number of clients (readers and writers) currently queued because of a lock on the MongoDB server. 
+
+### Locks
+ - **Acquired Locks**: The distribution of locks acquired and held by the MongoDB server. The following locks are monitored global_read, global_write, database_read, database_write, collection_read, collection_write.
+ ![image](https://user-images.githubusercontent.com/24860547/212876511-50384d07-98f0-4e37-8112-63634604903d.png)
+ 
+ - **Flow control events**: The timings of flow control events (acquiring and lagged) on the MongoDB server. 
```

Comment:
Chart snapshot missing. Do these metrics require special configuration?

Reply:
same comment as above - learn doc mentions no special configs, I think its just about what features are enabled in MongoDB.